lib/excon.rb

```diff
@@ -116,7 +116,7 @@ def new(url, params = {})
         :port       => uri.port,
         :query      => uri.query,
         :scheme     => uri.scheme
-      }.merge!(params)
```

Comment:
I think bang method is this case is not needed
For example if we create hash
```{}``` or ```Hash.new```  and merge! it , it is not matter what we use merge! or merge because {} in this case, is not a variable(local/instance/class etc), it is freshly created object(as a "value"), which is not assigned.
*Additional:
Travic CI failed becuase json failed, not because of my pull request.*
```
An error occurred while installing json (1.7.3), and Bundler cannot continue.
```